### ISSUE TITLE
Signal arg and Callback function param type missmatch nulls value without error

### ISSUE THREAD

**Issue by TheDuriel** [godotengine/godot#30170]
**Godot version:**

3.1

**Issue description:**

```
emit_signal("hey", my_dictionary)

func _on_hey(my_value: Directory) -> void: # Wrong type
    print(my_value)

>null
```

No error. No warning. Value simply becomes null and the code keeps running.
This should probably be caught as it can be difficult to notice.

**Comment by KoBeWi** [comment]
This would probably require types in signal definitions, which is already planned.

Also related (if not duplicate) to #24013

**Comment by KoBeWi** [comment]
Actually this was another issue which was fixed by #37318
